allennlp/training/callbacks/wandb.py

```diff
@@ -134,12 +127,31 @@ def _log(
             dict_to_log = {f"{log_prefix}/{k}": v for k, v in dict_to_log.items()}
         if epoch is not None:
             dict_to_log["epoch"] = epoch
-        self.wandb.log(dict_to_log, step=self.trainer._batch_num_total)  # type: ignore[union-attr]
+        self.wandb.log(dict_to_log, step=self.trainer._batch_num_total)  # type: ignore
 
     @overrides
     def on_start(
         self, trainer: "GradientDescentTrainer", is_primary: bool = True, **kwargs
     ) -> None:
         super().on_start(trainer, is_primary=is_primary, **kwargs)
+
+        if not is_primary:
+            return None
+
+        import wandb
+
+        self.wandb = wandb
+        self.wandb.init(**self._wandb_kwargs)
+
+        for fpath in self._files_to_save:
+            self.wandb.save(  # type: ignore
+                os.path.join(self.serialization_dir, fpath), base_path=self.serialization_dir
+            )
+
         if self._watch_model:
-            self.wandb.watch(self.trainer.model)  # type: ignore[union-attr]
+            self.wandb.watch(self.trainer.model)  # type: ignore
```

Comment:
Is this about `Item "None" of "Optional[Something]" has no attribute "watch"`? I have been fixing that with `assert self.wandb is not None`.

Reply:
It's because MyPy sees it as undefined, not `Optional`.